executor/join.go

```diff
@@ -312,8 +319,53 @@ func (e *HashJoinExec) handleJoinWorkerPanic(r interface{}) {
 	e.joinWorkerWaitGroup.Done()
 }
 
+func (e *HashJoinExec) handleUnmatchedRowsFromHashTable(workerID uint) (status bool, err error) {
+	var step = e.rowContainer.records.NumChunks() / int(e.concurrency)
+	var start = step * int(workerID)
+	var end = start + step
+	// The last worker handles all remaining chunks
+	if workerID+1 == e.concurrency {
+		end = e.rowContainer.records.NumChunks()
+	}
+	ok, joinResult := e.getNewJoinResult(workerID)
+	if !ok {
+		return false, err
+	}
+
+	for i := start; i < end; i++ {
+		chk := e.rowContainer.records.GetChunk(i)
+		for j := 0; j < chk.NumRows(); j++ {
+			if e.matchedStatus[i].UnsafeIsSet(j) == false { // process unmatched outer rows
+				e.joiners[workerID].onMissMatch(false, chk.GetRow(j), joinResult.chk)
+			}
+			if joinResult.chk.IsFull() {
+				e.joinResultCh <- joinResult
+				ok, joinResult = e.getNewJoinResult(workerID)
+				if !ok {
+					return false, err
+				}
+			}
+		}
+	}
+	if joinResult == nil {
+		return true, nil
+	} else if joinResult.err != nil || (joinResult.chk != nil && joinResult.chk.NumRows() > 0) {
+		e.joinResultCh <- joinResult
+	}
+	return true, nil
+}
+
 func (e *HashJoinExec) waitJoinWorkersAndCloseResultChan() {
 	e.joinWorkerWaitGroup.Wait()
+	// Concurrently handing unmatched rows from hash table at the tail
+	if e.outerHashJoin {
+		for i := uint(0); i < e.concurrency; i++ {
+			var workID = i
+			e.joinWorkerWaitGroup.Add(1)
+			go util.WithRecovery(func() { _, _ = e.handleUnmatchedRowsFromHashTable(workID) }, e.handleJoinWorkerPanic)
```

Comment:
The error returned from `handleUnmatchedRowsFromHashTable` is not handled.

Reply:
eliminate the returning info like `runJoinWorker`